keylime/mba/elparsing/tpm2_tools_elparser.py

```diff
@@ -166,11 +166,15 @@ def __unescape_eventlog(log: Dict) -> None:  # type: ignore
 
     def recursive_unescape(data):  # type: ignore
         if isinstance(data, str):
-            if data.startswith('"') and data.endswith('"'):
-                data = data[1:-1]
-                for orig, escaped in escaped_chars:
-                    data = data.replace(escaped, orig)
-                data = data.rstrip("\0")
+            if Version(tpm2_tools_version) < Version("5.6"):
```

Comment:
Do we need this, or this simpler version is equivalent?

```python
if data.startswith('"') and data.endswith('"'):
    data = data[1:-1]
    for orig, escaped in escaped_chars:
        data = data.replace(escaped, orig)
data = data.rstrip("\0")
```

If `data` is optional, we can add a guard before

Reply:
We still need that because otherwise we would strip intentional quotes on newer tpm2-tools versions,
but the data and rstrip (we also should only strip the exact last byte and not the all trailing ones) can be factored out.

Reply:
ah that makes sense

Reply:
@aplanas I now updated it. Have a look if it makes sense.